src/hypofuzz/hy.py

```diff
@@ -349,15 +349,34 @@ def _report(self, report: Report) -> None:
         db = get_db()
         db.save_metadata(self.database_key, report)
 
-        if (
-            self._last_report
-            and self._last_report["branches"] == report["branches"]
-            and self._last_report["note"] == report["note"]
-            and not self.pool.interesting_examples
+        # To avoid large db sizes, we only store some fields for the latest report.
+        # Additionally, if the previous report doesn't add anything relative to
+        # this one, we drop the previous report entirely.
+        #
+        # We first unconditionally drop the previous report. If we determine we
+        # should keep it, we then re-add its reduced version, with the fields we
+        # only keep for the latest report removed.
```

Comment:
Reading "first" here made me think it came before writing the latest report (above), so I'd probably rewrite this.  e.g.

> Having written the latest report, we can avoid bloating the database by dropping the previous report, and re-adding a trimmed version if it differs from the latest in more than just runtime.